src/Data/Text/Internal/Validate.hs

```diff
@@ -0,0 +1,155 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE UnliftedFFITypes #-}
+#if defined(SIMDUTF) || MIN_VERSION_bytestring(0,11,2)
+{-# LANGUAGE Trustworthy #-}
```

Comment:
`Data.Text.Encoding` used to promise `Trustworthy` unconditionally. Any particular reason not to follow its example?

I would rather remove `{-# LANGUAGE Trustworthy #-}` (not a fan of "Safe" Haskell) from both modules than promise it only conditionally, because it is likely to explode somewhere later in the ecosystem.

Reply:
I'll just remove it. I'm not even sure what guarantees Safe Haskell is supposed to provide, but it's trivial to trigger a segfault by misusing the functions exposed by this module.